node_modules/oae-authentication/tests/test-external-strategies.js

```diff
@@ -1293,4 +1293,165 @@ describe('Authentication', function() {
+            AuthenticationAPI.once(AuthenticationConstants.events.REFRESHED_STRATEGIES, function(tenant) {
+                return callback();
+            });
+        };
+
+        /**
+         * Mock the requests that the Passport library would make when verifying a user signing in through Google
+         *
+         * @param  {String}     email           The user's email address
+         */
+        var mockGoogleSignInRequests = function(email) {
+            // Require nock inline as it messes with the HTTP stack
+            // We only want this to happen in a controlled environment
+            var nock = require('nock');
+
+            // Ensure we can still perform regular HTTP requests during our tests
+            nock.enableNetConnect();
+
+            // Mock the "get access token" request in the OAuth2 cycle
+            var accessToken = _.random(10000);
+            nock('https://accounts.google.com')
+                .post('/o/oauth2/token')
+                .reply(200, {
+                    'access_token': accessToken,
+                    'refresh_token': 'foo'
+                });
+
+            // Mock the "get user profile" request
+            nock('https://www.googleapis.com')
+                .get('/plus/v1/people/me?access_token=' + accessToken)
+                .reply(200, {
+                    'kind': 'plus#person',
+                    'etag': 'RqKWnRU4WW46-6W3rWhLR9',
+                    'gender': 'male',
+                    'emails': [{'value': email, 'type': 'account'}],
+                    'urls': [
+                        {'value': 'http://www.youtube.com/user/abc123','type': 'otherProfile','label': 'ABC 123'},
+                    ],
+                    'objectType': 'person',
+                    'id': _.random(100000),
+                    'displayName': 'Foo Bar',
+                    'name': {
+                        'familyName': 'Bar',
+                        'givenName': 'Foo'
+                    },
+                    'url': 'https://plus.google.com/' + _.random(10000000),
+                    'image': {
+                        'url': 'https://lh5.googleusercontent.com/-wfVubfsOBV0/AAAAAAAAAAI/AAAAAAAAAGQ/rEb5FmsQuiA/photo.jpg?sz=50',
+                        'isDefault': false
+                    },
+                    'isPlusUser': true,
+                    'language': 'en',
+                    'verified': false
+                });
+        };
+
+        /**
+         * Assert that signing in through google with a certain email address does the right thing
+         *
+         * @param  {String}     email               The email address of the user that will sign in
+         * @param  {Boolean}    expectSuccess       Whether or not google authentication is expected to succeed
+         * @param  {Function}   callback            Standard callback function
+         */
+        var assertGoogleLogin = function(email, expectSuccess, callback) {
```

Comment:
This would be better in an `oae-authentication/lib/test/util.js` file. Something like:

`AuthenticationTestUtil.assertGoogleAuthenticationSucceeds(email, callback)` and `AuthenticationTestUtil.assertGoogleAuthenticationFails(email, <expected http code>, callback)`

The `<expected http code>` we could swap out for something that more accurately determines the kind of failure we were expecting (e.g., `reason`).